Sources/Schema.swift

```diff
@@ -100,7 +101,27 @@ extension SchemaBuilder {
         }
 
         let name = components[2]
+
+        if let cachedSchema = ReferenceResolver.cachedSchemaReferences[name] {
+            return Structure(name: name, structure: cachedSchema)
+        }
+        if let resolving = ReferenceResolver.resolvingReferences[name], resolving {
+            return Structure(name: name, structure: .unresolvedReference(pointer.path))
+        }
+        ReferenceResolver.resolvingReferences[name] = true
         let schema = try builder.build(swagger)
+        ReferenceResolver.cachedSchemaReferences[name] = schema
+        ReferenceResolver.resolvingReferences[name] = nil
         return Structure(name: name, structure: schema)
     }
 }
+
+enum ReferenceResolver {
```

Comment:
Yeah i just wrote that in the GitHub text window haha
I'm making a branch and cleaning it up right now.